app/controllers/submission_comments_api_controller.rb

```diff
@@ -18,12 +18,84 @@
 
 # @API Submission Comments
 #
-# This API can be used to create files to attach to submission comments.  The
+# This API can be used to edit and delete submission comments.
+# It can also be used to create files to attach to submission comments. The
 # submission comments themselves can be created using the Submissions API.
 class SubmissionCommentsApiController < ApplicationController
   before_action :require_context
 
   include Api::V1::Attachment
+  include Api::V1::SubmissionComment
+
+  # @API Edit a submission comment
+  #
+  # Edit the given submission comment.
+  #
+  # @argument comment [String]
+  #   If this argument is present, edit thet text of a comment.
+  #
+  # @argument attachment_ids[] [Integer]
+  #   Attach files to this comment that were previously uploaded using the
+  #   Submission Comment API's files action.
+  #
+  # @argument media_comment_id [String]
+  #   Add an audio/video comment to the submission. Media comments can be added
+  #   via this API, however, note that there is not yet an API to generate or
+  #   list existing media comments, so this functionality is currently of
+  #   limited use.
+  #
+  # @argument media_comment_type [String, "audio"|"video"]
+  #   The type of media comment being added.
+  #
+  # @returns SubmissionComment
+  def update
+    submission_comment = SubmissionComment.find(params[:id])
+    if authorized_action(submission_comment, @current_user, :update)
+      submission_comment.updating_user = @current_user
+      submission_comment.reload unless submission_comment.update(submission_comment_params)
+
+      return render json: anonymous_moderated_submission_comments_json(
```

Comment:
This doesn't need to go through the anonymous/moderated filter because the current_user has permission to update the comment, which means they own it (and thus, know their own identity). So we don't need to be wary of hiding author_id and such.

The use on line 84 is warranted, though, since a user who is not the author can delete a submission comment.